### ISSUE TITLE
The detection results on val split is poor.

### ISSUE THREAD

**Issue by jichaofeng** [xinzhuma/monodle#13]
I train the model on single GPU and never use the distributed training.  The batch size is set at 8. The detection results are as follows:
Car AP@0.70, 0.70, 0.70:
bbox AP:89.6069, 86.3804, 78.2012
bev  AP:23.5712, 18.8870, 15.9993
3d   AP:16.8725, 13.4053, 12.3370
aos  AP:88.85, 84.82, 76.00
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:94.9005, 89.4963, 80.5794
bev  AP:21.5961, 16.3269, 13.8877
3d   AP:14.9669, 11.0160, 9.5246
aos  AP:93.97, 87.69, 78.04

**Comment by mrsempress** [comment]
Did you get warning `W IndexingUtils.h:20] Warning: indexing with dtype torch.uint8 is now deprecated, please
use a dtype torch.bool instead. (function expandTensors)` ? I don't know where use `torch.uint8`.
I get even worse results: (maybe because of this warning)
```
Car AP@0.70, 0.70, 0.70:
bbox AP:87.7941, 85.7247, 77.1482
bev  AP:17.1880, 14.6495, 12.3512
3d   AP:11.6470, 10.2226, 9.4794
aos  AP:86.73, 84.40, 75.38
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:92.8573, 87.3384, 77.7636
bev  AP:15.8635, 12.1556, 10.3563
3d   AP:10.0447, 7.4823, 6.6572
aos  AP:91.63, 85.93, 75.91
```

**Comment by xinzhuma** [comment]
hi @jichaofeng @mrsempress, can you post the config file? I will check it.

**Comment by xinzhuma** [comment]
@mrsempress this warning is caused by the mismatch of the pytorch versions. Our code is written under pytorch1.1, some functions used in this repo may be deprecated in the higher versions. Please set up your environment according to the README.

**Comment by jichaofeng** [comment]
Thank you for your answer. The program is running without any warning. This is the content of kitti_example.yaml:
random_seed: 444
dataset:
  type: &dataset_type 'KITTI'
  batch_size: 8
  use_3d_center: True
  class_merging: False
  use_dontcare: False
  bbox2d_type: 'anno'   # 'proj' or 'anno'
  meanshape: False      # use predefined anchor or not
  writelist: ['Car']
  random_flip: 0.5
  random_crop: 0.5
  scale: 0.4
  shift: 0.1
model:
  type: 'centernet3d'
  backbone: 'dla34'
  neck: 'DLAUp'
  num_class: 3
optimizer:
  type: 'adam'
  lr: 0.00125
  weight_decay: 0.00001
lr_scheduler:
  warmup: True  # 5 epoches, cosine warmup, init_lir=0.00001 in default
  decay_rate: 0.1
  decay_list: [90, 120]
trainer:
  max_epoch: 140
  gpu_ids: 0
  save_frequency: 10 # checkpoint save interval (in epoch)
  # resume_model: 'checkpoints/checkpoint_epoch_70.pth'
tester:
  type: *dataset_type
  mode: single   # 'single' or 'all'
  checkpoint: 'checkpoints/checkpoint_epoch_140.pth'  # for 'single' mode
  checkpoints_dir: 'checkpoints'  # for 'all' model
  threshold: 0.2  # confidence filter

**Comment by jichaofeng** [comment]
Thank you for your answer. The programe is running without any warning. This is the content of kitti_example.yaml:
random_seed: 444
dataset:
  type: &dataset_type 'KITTI'
  batch_size: 8
  use_3d_center: True
  class_merging: False
  use_dontcare: False
  bbox2d_type: 'anno'   # 'proj' or 'anno'
  meanshape: False      # use predefined anchor or not
  writelist: ['Car']
  random_flip: 0.5
  random_crop: 0.5
  scale: 0.4
  shift: 0.1
model:
  type: 'centernet3d'
  backbone: 'dla34'
  neck: 'DLAUp'
  num_class: 3

optimizer:
  type: 'adam'
  lr: 0.00125
  weight_decay: 0.00001
lr_scheduler:
  warmup: True  # 5 epoches, cosine warmup, init_lir=0.00001 in default
  decay_rate: 0.1
  decay_list: [90, 120]

trainer:
  max_epoch: 140
  gpu_ids: 0
  save_frequency: 10 # checkpoint save interval (in epoch)
  # resume_model: 'checkpoints/checkpoint_epoch_70.pth'
tester:
  type: *dataset_type
  mode: single   # 'single' or 'all'
  checkpoint: 'checkpoints/checkpoint_epoch_140.pth'  # for 'single' mode
  checkpoints_dir: 'checkpoints'  # for 'all' model
  threshold: 0.2  # confidence filter

**Comment by jichaofeng** [comment]
Thank you for your answer. The programe is running without any warning. This is the content of kitti_example.yaml:
random_seed: 444

dataset:
  type: &dataset_type 'KITTI'
  batch_size: 8
  use_3d_center: True
  class_merging: False
  use_dontcare: False
  bbox2d_type: 'anno'   # 'proj' or 'anno'
  meanshape: False      # use predefined anchor or not
  writelist: ['Car']
  random_flip: 0.5
  random_crop: 0.5
  scale: 0.4
  shift: 0.1

model:
  type: 'centernet3d'
  backbone: 'dla34'
  neck: 'DLAUp'
  num_class: 3

optimizer:
  type: 'adam'
  lr: 0.00125
  weight_decay: 0.00001

lr_scheduler:
  warmup: True  # 5 epoches, cosine warmup, init_lir=0.00001 in default
  decay_rate: 0.1
  decay_list: [90, 120]

trainer:
  max_epoch: 140
  gpu_ids: 0
  save_frequency: 10 # checkpoint save interval (in epoch)
  # resume_model: 'checkpoints/checkpoint_epoch_70.pth'
tester:
  type: *dataset_type
  mode: single   # 'single' or 'all'
  checkpoint: 'checkpoints/checkpoint_epoch_140.pth'  # for 'single' mode
  checkpoints_dir: 'checkpoints'  # for 'all' model
  threshold: 0.2  # confidence filter

**Comment by jichaofeng** [comment]
Thank you for your answer. The programe is running without any warning. This is the content of kitti_example.yaml:
random_seed: 444

dataset:
  type: &dataset_type 'KITTI'
  batch_size: 8
  use_3d_center: True
  class_merging: False
  use_dontcare: False
  bbox2d_type: 'anno'   # 'proj' or 'anno'
  meanshape: False      # use predefined anchor or not
  writelist: ['Car']
  random_flip: 0.5
  random_crop: 0.5
  scale: 0.4
  shift: 0.1

model:
  type: 'centernet3d'
  backbone: 'dla34'
  neck: 'DLAUp'
  num_class: 3

optimizer:
  type: 'adam'
  lr: 0.00125
  weight_decay: 0.00001

lr_scheduler:
  warmup: True  # 5 epoches, cosine warmup, init_lir=0.00001 in default
  decay_rate: 0.1
  decay_list: [90, 120]

trainer:
  max_epoch: 140
  gpu_ids: 0
  save_frequency: 10 # checkpoint save interval (in epoch)
tester:
  type: *dataset_type
  mode: single   # 'single' or 'all'
  checkpoint: 'checkpoints/checkpoint_epoch_140.pth'  # for 'single' mode
  checkpoints_dir: 'checkpoints'  # for 'all' model
  threshold: 0.2  # confidence filter

**Comment by jichaofeng** [comment]
![2021-09-08 14-28-41屏幕截图](https://user-images.githubusercontent.com/21376860/132457606-90c02250-9500-4daf-988e-0db806af8c4f.png)
I change the program as shown the figure to train on single GPU.

**Comment by xinzhuma** [comment]
@jichaofeng it seems okay. I am not sure whether you used the lasted version, maybe you need to check it. Besides, you need to adjust the learning rate according to the linear rule if the batch size is changed.

**Comment by jichaofeng** [comment]
Thank you for your advise. @xinzhuma  I adjust the learning rate and set it as 0.0004. The results turn better.
INFO  Car AP@0.70, 0.70, 0.70:
bbox AP:97.2279, 88.5018, 80.0014
bev  AP:27.5880, 23.7416, 22.2742
3d   AP:19.5566, 17.9170, 17.0286
aos  AP:96.47, 87.47, 78.46
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:98.0585, 92.0243, 84.9332
bev  AP:24.8477, 19.2732, 16.7192
3d   AP:17.3024, 13.1625, 11.6137
aos  AP:97.29, 90.86, 83.07

**Comment by xinzhuma** [comment]
Great, and I will close this issue.
